transport/rest/src/main/java/io/a2a/transport/rest/handler/RestHandler.java

```diff
@@ -0,0 +1,433 @@
+    }
+
+    private Flow.Publisher<String> convertToSendStreamingMessageResponse(
+            Flow.Publisher<StreamingEventKind> publisher) {
+        // We can't use the normal convertingProcessor since that propagates any errors as an error handled
+        // via Subscriber.onError() rather than as part of the SendStreamingResponse payload
+        return ZeroPublisher.create(createTubeConfig(), tube -> {
+            publisher.subscribe(new Flow.Subscriber<StreamingEventKind>() {
+                Flow.Subscription subscription;
+
+                @Override
+                public void onSubscribe(Flow.Subscription subscription) {
+                    this.subscription = subscription;
+                    subscription.request(1);
+                }
+
+                @Override
+                public void onNext(StreamingEventKind item) {
+                    try {
+                        String payload = JsonFormat.printer().omittingInsignificantWhitespace().print(ProtoUtils.ToProto.taskOrMessageStream(item));
+                        System.out.println("############## Sending event " + payload);
+                        tube.send(payload);
+                        subscription.request(1);
+                    } catch (InvalidProtocolBufferException ex) {
+                        onError(ex);
+                    }
+                }
+
+                @Override
+                public void onError(Throwable throwable) {
+                    System.out.println("############## Sending error " + throwable);
+                    throwable.printStackTrace();
+                    if (throwable instanceof JSONRPCError jsonrpcError) {
+                        tube.send(new HTTPRestErrorResponse(jsonrpcError).toJson());
+                    } else {
+                        tube.send(new HTTPRestErrorResponse(new InternalError(throwable.getMessage())).toJson());
+                    }
+                    onComplete();
+                }
+
+                @Override
+                public void onComplete() {
+                    tube.complete();
+                }
+            });
+        });
+    }
+
+    private int mapErrorToHttpStatus(JSONRPCError error) {
+        if (error instanceof InvalidRequestError || error instanceof InvalidParamsError) {
+            return 400;
+        }
+        if (error instanceof MethodNotFoundError || error instanceof TaskNotFoundError) {
+            return 404;
+        }
+        if (error instanceof TaskNotCancelableError || error instanceof PushNotificationNotSupportedError
+                || error instanceof UnsupportedOperationError) {
+            return 501;
+        }
+        if (error instanceof ContentTypeNotSupportedError) {
+            return 415;
+        }
+        if (error instanceof InternalError || error instanceof InvalidAgentResponseError) {
+            return 500;
+        }
+        return 500;
+    }
+
+    public AgentCard getAgentCard() {
+        return agentCard;
```

Comment:
With above change related to the extended agent card, I think this could then be updated to return the extended agent card similar to this:

https://github.com/a2aproject/a2a-python/blob/main/src/a2a/server/apps/rest/rest_adapter.py#L242-L245